salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                        log.debug('Secrets engine "%s" at path "%s" is disabled.',
+                                secret_engine.type, secret_engine.type)
+        except Exception:
+            raise
+
+        if has_changes:
+            ret['changes']['new'] = ["type: {} - path: {}".format(ob.type, ob.path) for ob in local_engines]
+
+        log.info('Finished cleaning up secrets engines.')
+
+
+class VaultAuditManager():
+    """
+    Module for handling Vault Audit Devices
+    """
+
+    def __init__(self):
+        """Initialize Vault Audit Managers
+        """
+        log.info("Initializing Vault Audit Manager...")
+
+    def get_remote_audit_devices(self, client, ret):
+        """Get audit devices information from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- audit devices
+        """
+        log.info("Retrieving audit devices from vault...")
+        devices = []
+        try:
+            audit_devices_resp = client.sys.list_enabled_audit_devices()
+            log.debug(audit_devices_resp)
+            for device in audit_devices_resp['data']:
+                audit_device = audit_devices_resp[device]
+                devices.append(
+                    VaultAuditDevice(
+                        type=audit_device['type'],
+                        path=(audit_device["path"]
+                              if 'path' in audit_device else device),
+                        description=audit_device["description"],
+                        config=OrderedDict(
+                            sorted(audit_device["options"].items()))
+                    )
+                )
+
+            log.info('Finished retrieving audit devices from vault.')
+        except Exception:
+            raise
+
+        return devices
+
+    def get_local_audit_devices(self, configs, ret):
+        """Get audit device inforamtion from local config file
+
+        Arguments:
+            configs {list} -- audit devices
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- audit devices
+        """
+        log.info("Loading audit devices from local config...")
+        devices = []
+        if configs:
+            try:
+                for audit_device in configs:
+                    config = None
+                    if 'config' in audit_device:
+                        if audit_device['config']:
+                            config = OrderedDict(
+                                sorted(audit_device["config"].items()))
+
+                    devices.append(
+                        VaultAuditDevice(
+                            type=audit_device["type"],
+                            path=audit_device["path"],
+                            description=audit_device["description"],
+                            config=config
+                        )
+                    )
+            except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.